lib/new_relic/telemetry/plug.ex

```diff
@@ -0,0 +1,184 @@
+defmodule NewRelic.Telemetry.Plug do
+  use GenServer
+
+  @moduledoc """
+  `Plug` based HTTP servers are auto-instrumented based on the `telemetry` integration
+  built into `Plug.Cowboy` and `Plug`.
+
+  To prevent reporting the current transaction, call:
+  ```elixir
+  NewRelic.ignore_transaction()
+  ```
+
+  Inside a Transaction, the agent will track work across processes that are spawned as
+  well as work done inside a Task Supervisor. When using `Task.Supervisor.async_nolink`
+  you can signal to the agent not to track the work done inside the Task, which will
+  exclude it from the current Transaction. To do this, send in an additional option:
+
+  ```elixir
+  Task.Supervisor.async_nolink(
+    MyTaskSupervisor,
+    fn -> do_work() end,
+    new_relic: :no_track
+  )
+  ```
+  """
+
+  alias NewRelic.{Transaction, DistributedTrace, Util}
+
+  # TODO: Should Telemetry GenServers only be started if NewRelic.Config.enabled?()
```

Comment:
What do you think about leaving this logic only to `NewRelic.Telemetry.Supervisor`?

This should cover the requirement and avoid duplication - I can see this pattern becoming very repetitive as more telemetry instrumentation get added.

Reply:
I'll probably just move `TelemetrySupervisor` under `EnabledSupervisor` which already does this logic...